Add KS30 "The Scarlet King"

Skold177 opened the pull request:
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

This pull request adds the KS30 Battlefield "The Scarlet King"

## Steps to test these changes

Go to Chamber of Oracles and test the KS30 with an Atropos Orb

## Captures ##

This KS30 was created based off of this capture.

https://www.dropbox.com/scl/fi/sjg76758e6rbziw9fsaw2/KS30_The_Scarlet_King.zip?rlkey=bnmacmlb48lj0je0jhtwpa09h&st=92r1bhvu&dl=0

https://www.youtube.com/watch?v=KT6otcqBFYg

Skold177 committed:
Add KS30 "The Scarlet King"

This adds the KS30 Battlefield "The Scarlet King" based off retail captures.

Skold177 committed:
Merge branch 'LandSandBoat:base' into KS30-The-Scarlet-King

CriticalXI commented on scripts/zones/Chamber_of_Oracles/mobs/Purson.lua:
```diff
@@ -0,0 +1,44 @@
+end
+
+entity.onMobEngage = function(mob, target)
+    mob:setLocalVar('BloodWeaponTimer', GetSystemTime())
```
Should add a 15 second delay on engage to this here going by your capture.

MowFord replied:
Also might be based on hpp. I didn't watch the video

CriticalXI replied:
They hadn't done any damage when it used blood weapon the first time.

CriticalXI commented on scripts/zones/Chamber_of_Oracles/mobs/Purson.lua:
```diff
@@ -0,0 +1,44 @@
+
+    -- Set Modifiers
+entity.onMobSpawn = function(mob)
+    mob:setMagicCastingEnabled(false) -- Does not cast spells.
```
Instead of setting this on lua, an alternative would be to remove its spell list in mob_pools..sql so it has no spells to cast.

CriticalXI reviewed (commented):
Looks good to me other than a couple minor adjustments. I couldn't find anything missing behavior wise from the capture.

WinterSolstice8 commented on scripts/enum/mob_skills.lua:
```diff
@@ -84,6 +84,8 @@ xi.mobSkill =
 
     FAMILIAR_1               =  740, -- "Tango with a Tracker" Shikaree X
 
+    GREAT_WHIRLWIND          =  803, -- "The Scarlet King" KSNM 30 (Purson)
```
Comment is probably unnecessary here if it matches standard great whirlwinds from other manticores

Xaver-DaRed replied:
This should be GREAT_WHIRLWIND_1 since there are other mobskills named the same

Xaver-DaRed replied:
Also, please do not resolve conversations, thats for reviewers to do.
<img width="797" height="32" alt="imagen" src="https://github.com/user-attachments/assets/e87dd706-052b-4920-b3bc-c4bf914c45a7" />

Skold177 replied:
Accident on my end, thanks for the review - this enum change has been updated. Thank you!

Skold177 committed:
Updated KS30 The Scarlet King BW

Added delay before Blood Weapon begins and fixed newline issue

Xaver-DaRed reviewed (changes requested):
Change the name of the mobskill to indicate has a copy/copies

Skold177 committed:
Update KS30 Great_Whirlwind enum

Changes GREAT_WHIRLWIND in the mob_skills.lua to GREAT_WHIRLWIND_1 since there are multiple versions of this skill
